portability/memory.cc

```diff
@@ -403,7 +403,7 @@ toku_xcalloc(size_t nmemb, size_t size) {
 
 void *
 toku_xrealloc(void *v, size_t size) {
-#if __APPLE__
+#if defined(__APPLE__) && __APPLE__
```

Comment:
just if defined or ifdef, no need to check the actual value of the macro, according to the docs (https://developer.apple.com/library/content/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-SW13)